packages/shared/lib/services/connection.service.ts

```diff
@@ -438,67 +404,14 @@ class ConnectionService {
-                | AppCredentials
-                | OAuth2ClientCredentials
-                | TableauCredentials
-                | JwtCredentials
-                | TwoStepCredentials
-                | BillCredentials
-                | SignatureCredentials;
-            if (credentials.type && credentials.type === 'OAUTH2') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'APP') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'OAUTH2_CC') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'TABLEAU') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'JWT') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'BILL') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'SIGNATURE') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
-
-            if (credentials.type && credentials.type === 'TWO_STEP') {
-                const creds = credentials;
-                creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
-                connection.credentials = creds;
-            }
+        const credentials = connection.credentials;
+        if (credentials.type && 'expires_at' in credentials) {
+            const creds = credentials;
+            creds.expires_at = creds.expires_at != null ? parseTokenExpirationDate(creds.expires_at) : undefined;
+            connection.credentials = creds;
         }
```

Comment:
I have re-read this section 20times, I can't understand why we copied the code. But please let me know

Reply:
I believe it may be a good time to refractor it, like we did with the `upsertConnection` functions.